pkg/apis/cstor/v1/cstorrestore.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=cstorrestore
+
+// CStorRestore describes a cstor restore resource created as a custom resource
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,shortName=crestore
+// +kubebuilder:printcolumn:name="Backup",type=string,JSONPath=`.spec.restoreName`,description="Name of the snapshot which is restored"
+// +kubebuilder:printcolumn:name="Volume",type=string,JSONPath=`.spec.volumeName`,description="Volume on which restore is performed"
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status`,description="Identifies the state of the restore"
+type CStorRestore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"` // set name to restore name + volume name + something like cspi tag
+	Spec              CStorRestoreSpec            `json:"spec"`
+	Status            CStorRestoreStatus          `json:"status,omitempty"`
```

Comment:
internalapi dir is not in sync with the latest change in apis

Reply:
like missing json tag and type comments

Reply:
Updates PTAL